dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java

```diff
@@ -75,11 +97,126 @@ public static void prepare() throws Exception {
+        final String rootFolderName = String.format("lolFolder-%d", System.currentTimeMillis());
+        final Folder root1 = folderAPI.createFolders(rootFolderName, host, user, false);
+        final FileAsset fileAsset = new FileAsset();
+        final ImmutableFileAssetContentType.Builder builder = ImmutableFileAssetContentType.builder();
+        builder.name("Test").variable("testfa");
+        final ContentType fileAssetContentType = contentTypeAPI.find("FileAsset");
+        final String fileName1 = TEMP_FILE + System.currentTimeMillis();
+        final File tempFile1 = File.createTempFile(fileName1, TXT);
+        final String anyContent = "LOL!";
+        FileUtil.write(tempFile1, anyContent);
+        final String fileNameField1 = fileName1 + DOT_TXT;
+        final String title1 = "Contentlet-1";
+
+        fileAsset.setContentType(fileAssetContentType);
+        fileAsset.setFolder(root1.getInode());
+        fileAsset.setBinary(FileAssetAPI.BINARY_FIELD, tempFile1);
+        fileAsset.setStringProperty(FileAssetAPI.HOST_FOLDER_FIELD, root1.getInode());
+        fileAsset.setStringProperty(FileAssetAPI.TITLE_FIELD, title1);
+        fileAsset.setStringProperty(FileAssetAPI.FILE_NAME_FIELD, fileNameField1);
+        fileAsset.setIndexPolicy(IndexPolicy.FORCE);
+
+        // Create a piece of content for the default host
+        final Map<String,Object>  contentletMap = esMappingAPI.toMap(APILocator.getContentletAPI().checkin(fileAsset, user, false));
+
+        assertNotNull(contentletMap);
+        assertEquals(fileNameField1.toLowerCase(), contentletMap.get("fileasset.filename"));
+        assertEquals("fileasset", contentletMap.get("structurename"));
+        assertEquals("text/plain; charset=iso-8859-1", contentletMap.get("metadata.contenttype"));
+        assertEquals("4", contentletMap.get("metadata.filesize"));
+        assertTrue( contentletMap.get("metadata.content").toString().contains("lol!"));
+
+    }
+
+    @Test
+    public void test_toMap_binary_field_shouldSuccess() throws Exception {
+
+        final ESMappingAPIImpl esMappingAPI    = new ESMappingAPIImpl();
+        final FieldAPI         fieldAPI        = APILocator.getContentTypeFieldAPI();
+        final Host host = APILocator.getHostAPI().findDefaultHost(user, false);
+        final String varname = "testcontenttypetwobinaryfields" + System.currentTimeMillis();
+
+        ContentType contentType = ContentTypeBuilder.builder(BaseContentType.CONTENT.immutableClass())
+                .description("Test ContentType Binary Fields")
+                .host(host.getIdentifier())
+                .name("Test ContentType Binary Fields")
+                .owner("owner")
+                .variable(varname)
+                .build();
+
+        contentType = contentTypeAPI.save(contentType);
+
+        Field textField = ImmutableTextField.builder()
+                .name("Title")
+                .variable("title")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "title" appears 6 times in this file; the first occurrence is on line 164](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)